Cue/app/tabs/library/DeckView.js

```diff
@@ -92,6 +94,41 @@ class DeckView extends React.Component {
     this.props.flagCard(this.props.deckUuid, cardUuid, flag)
   }
 
+  _rateDeck = () => {
+    let buttons = [
+      {text: 'Recommend',  onPress: () => this.props.rateDeck(this.state.deck.uuid, 1)},
+      {text: 'Would not Recommend', onPress: () => this.props.rateDeck(this.state.deck.uuid, -1)},
+      {text: 'Cancel', style: 'cancel'},
+    ]
+
+    Alert.alert(
+      (Platform.OS === 'android'
+        ? 'Rate this deck'
+        : 'Rate This Deck'),
+      'Would you recommend this deck to others?',
+      (Platform.OS === 'android'
+        ? buttons.reverse()
+        : buttons)
+    )
+  }
+
+  _copyDeck = () => {
+    Alert.alert(
+      (Platform.OS === 'android'
+        ? 'Copy this deck?'
+        : 'Copy This Deck?'),
+      'Copying will create a private version of this deck in your library.'
+        + ' The copied deck will no longer receive updates from the original owner.',
+      [
+        {text: 'Cancel', style: 'cancel'},
+        {text: 'Copy', onPress: () => {
+          let action = this.props.copyDeck(this.state.deck)
+          this.props.navigator.push({deck: action.deck})
```

Comment:
I would do a `replace` here actually so that tapping the back button after copying the deck returns the user to the Library screen rather than to the shared / public deck.